From 9ae0b23e4875ba44a2baa0aa91ef8f9629b3beb3 Mon Sep 17 00:00:00 2001
From: Orgad Shaneh <orgad.shaneh@audiocodes.com>
Date: Mon, 19 Aug 2024 11:54:09 +0300
Subject: [PATCH 1/3] pcaputil: Use clock rate and channel count from coder

For opus, the input channel count is 2 (stereo), but the coder initializes
it to 1, and the actual data has 1 channel.
---
 pjsip-apps/src/samples/pcaputil.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/pjsip-apps/src/samples/pcaputil.c b/pjsip-apps/src/samples/pcaputil.c
index d1621fabb..d41bdc753 100644
--- a/pjsip-apps/src/samples/pcaputil.c
+++ b/pjsip-apps/src/samples/pcaputil.c
@@ -328,15 +328,15 @@ static void pcap2wav(const pj_str_t *codec,
     T( pjmedia_codec_open(app.codec, &param) );
 
     /* Init audio device or WAV file */
-    samples_per_frame = ci->clock_rate * param.info.frm_ptime / 1000;
+    samples_per_frame = param.info.clock_rate * param.info.frm_ptime / 1000;
     if (pj_strcmp2(wav_filename, "-") == 0) {
         pjmedia_aud_param aud_param;
 
         /* Open audio device */
         T( pjmedia_aud_dev_default_param(dev_id, &aud_param) );
         aud_param.dir = PJMEDIA_DIR_PLAYBACK;
-        aud_param.channel_count = ci->channel_cnt;
-        aud_param.clock_rate = ci->clock_rate;
+        aud_param.channel_count = param.info.channel_cnt;
+        aud_param.clock_rate = param.info.clock_rate;
         aud_param.samples_per_frame = samples_per_frame;
         T( pjmedia_aud_stream_create(&aud_param, NULL, &play_cb, 
                                      NULL, &app.aud_strm) );
@@ -344,7 +344,7 @@ static void pcap2wav(const pj_str_t *codec,
     } else if (pj_stristr(wav_filename, &WAV)) {
         /* Open WAV file */
         T( pjmedia_wav_writer_port_create(app.pool, wav_filename->ptr,
-                                          ci->clock_rate, ci->channel_cnt,
+                                          param.info.clock_rate, param.info.channel_cnt,
                                           samples_per_frame,
                                           param.info.pcm_bits_per_sample, 0, 0,
                                           &app.wav) );
-- 
2.43.0

